interview-questions.md

```diff
@@ -8,42 +8,49 @@ Then, research the question to expand on your answer. Even if you feel you have
 
 1. What is object-oriented programming? How is it different than functional programming?
 
-Your answer:
+Your answer:Object-oriented programming is when coding is bases its foundations on coding around the object to make fuctional code. Such as creating a a method and bringing in object to perform in that method. Whereas fuctional programming is the process of producing code to have a specific behavior or perform a certain task. An example is creating a function and passing through arguments or data to perform the behavior or task it has been created for, like removing odd numbers in array where the array is the data inputted into a function and produces only even numbers.
 
-Researched answer:
+Researched answer: Object Oriented Programming are classes and objects. It is structured for simple reusable code. This is a good style of coding because it is can be dynamically used such as coding for movies: run time, title, year made. This code would be able to produce over all movies rather than being restricted to one. The four building blocks are Classes, Objects, Methods and Attributes.
 
 1. What is the difference between a Float and an Integer in Ruby?
 
-Your answer:
+Your answer:In Red Robin Ruby a Float and Integer are similar in the aspects of them both being numerical data and where they differ is Float will produce a decimal value and an Integer will give a a whole value with NO decimals.
 
-Researched answer:
+Researched answer:A float is a number or a representation number for irrationial numbers and creates a decimal point. The inter is the opposit of float and does not have a fractional part and is a whole number.
 
 1. Ruby has an implicit return. What does that mean?
 
-Your answer:
+Your answer:Ruby does not have a "return" so instead it has implicit return which will always produce the last line of method and its value. An auto return of last value.
 
-Researched answer:
+Researched answer: Implicit return of a Ruby method is the final statement in the method. Which Ruby makes the developers work easier by doing this if we want to know how our Ruby method works.
 
 1. What is a block in Ruby?
 
-Your answer:
+Your answer:Block is the code between the def and end.  The body of the logic being produced. For example: def=start of the method, end= end of the method and *=area of block
 
```

Comment:
def and end are going to signal a method. A block in Ruby is within a do and an end.